psydac/cad/geometry.py

```diff
@@ -326,3 +327,216 @@ def export( self, filename ):
+    geom = yaml.dump( data   = topo_yml,
+                     Dumper = yamlloader.ordereddict.Dumper )
+    # Write topology metadata as fixed-length array of ASCII characters
+    h5['topology.yml'] = np.array( geom, dtype='S' )
+
+    group = h5.create_group( yml['patches'][i]['mapping_id'] )
+    group.attrs['degree'     ] = patch.degree
+    group.attrs['rational'   ] = rational
+    group.attrs['periodic'   ] = tuple( False for d in range( patch.dim ) ) if periodic is None else periodic
+    for d in range( patch.dim ):
+        group['knots_{}'.format( d )] = patch.knots[d]
+    group['points'] = patch.points
+    if rational:
+        group['weights'] = patch.weights
+
+    h5.close()
+
+#==============================================================================
+def refine(nrb, ncells=None, degree=None , periodic=None):
+
+    nrb = nrb.clone()
+    if ncells is not None:
+
+        for axis in range(0,nrb.dim):
+            ub = nrb.knots[axis][0]
+            ue = nrb.knots[axis][-1]
+            knots = list(np.linspace(ub,ue,ncells[axis]+2)[1:-1])
+            for k in nrb.knots[axis]:
+                if k in knots:
+                    knots.remove(k)
+            nrb.refine(axis, knots)
+
+    if degree is not None:
+        for axis in range(0,nrb.dim):
+            d = degree[axis] - nrb.degree[axis]
+            nrb.elevate(axis, times=d)
+
+    return nrb
+
+#==============================================================================
+def create_geometry_file(nrb, filename, ncells, degree, periodic=None):
+
+    import igakit
+    assert isinstance(nrb, igakit.nurbs.NURBS)
+    import os.path
+
+    nrb = refine(nrb, ncells=[i-1 for i in ncells], degree=degree)
+    # ...
+
+    extension = os.path.splitext(filename)[1]
+    if not extension == '.h5':
+        raise ValueError('> Only h5 extension is allowed for filename')
+
+    export_geo(filename, nrb, periodic=periodic)
+
+#==============================================================================
+class geopdes(object):
+    def __init__(self):
+        self._list_begin_line = []
+
+        self._n_lines_per_patch = 0
+
+    def read(self, filename):
+
+        f = open(filename)
+        lines = f.readlines()
+        f.close()
+
+        _lines = []
+        for line in lines:
+            if line[0].strip() != "#":
+                _lines.append(line)
+        lines = _lines
+        data = self._read_header(lines[0])
+        n_dim = data[0]
+        r_dim = data[1]
+        n_patchs = data[2]
+
+        self._n_lines_per_patch = n_dim + n_dim + n_dim + 1
+
+        self._list_begin_line = self._get_begin_line(lines, n_patchs)
+
+
+        nrb = self._read_patch(lines, 1)
+
+        return nrb
+
+    def _read_header(self, line):
+        chars = line.split(" ")
+        data  = []
+        for c in chars:
+            try:
+                data.append(int(c))
+            except:
+                pass
+        return data
```

Comment:
This method does not use `self`, so it should be a `@staticmethod`. Or just a separate function.

Reply:
I turned it into a function